### ISSUE TITLE
Typecheck for argument/dictionary field default values

### ISSUE THREAD

**Issue by saschanaz** [w3c/webidl2.js#427]
```webidl
interface mixin Mixin {
  void op(boolean arg = 3); // no!
};

dictionary Dict {
  DOMString str = true; // no!
};
```

See also https://github.com/heycam/webidl/pull/750#issuecomment-536238161.

**Comment by TG1999** [comment]
Hi I will like to work on this issue, can you explain how can I solve this issue ?

**Comment by saschanaz** [comment]
I think this one is rather complex because we should make sure typedefs and unions are correctly examined. We'll want to modify `*validate()` in `lib/productions/argument.js` and check whether the argument type allows the value type.

**Comment by TG1999** [comment]
Okay thanks can you suggest me some issues for first timers

**Comment by saschanaz** [comment]
#220 seems easy enough if you know how modules work in modern JavaScript. 👍 We want to export `WebIDLParseError` so that package users can import it.

**Comment by TG1999** [comment]
Yeah I know how to export modules can you tell the filename so I can send a PR for it ASAP. :)

**Comment by saschanaz** [comment]
It's in `lib/tokeniser.js`, we want to export it to and again from `index.js`.

**Comment by TG1999** [comment]
Any more issues for first timers :smile:

**Comment by TG1999** [comment]
@saschanaz Any more issues for first-timers, I will also like to work on any new feature if you can guide me.

**Comment by saschanaz** [comment]
I suggest #415 where we need a webpack configuration to turn off class name mangling. This would be more of searching documentations and possibly also StackOverflow, and then writing some test.

The error object we exported in #428 should have `err.constructor.name === "WebIDLParseError"` but currently #415 makes it `_`, so we want to fix it.

**Comment by TG1999** [comment]
I am not able to get how to get started with it, some guidance will be surely helpful for me in this case.

**Comment by TG1999** [comment]
Hi @saschanaz any more issues on which I can work or any new feature which I can make for webidl2.js if you can guide me I can surely make some good features for webidl2.js and will surely learn something new while making the features.

**Comment by TG1999** [comment]
I will like to learn the working of web standards and contribute in it so will like to do anything related to it. I will read the docs in the meantime and come back to you when done.
